centreon-archive/centreon-archived
# [PoC] feat(UI): Implementation of multi columns dashboarding system

## Description

**This is a Proof of Concept**

https://user-images.githubusercontent.com/12515407/195647574-977245d2-894a-4de3-bbb0-a1535ff832db.mov

This PoC evaluates the complexity of developing a multi columns dashboarding system.

Here are the features supported by this PoC:
- Add a system of 3 columns (will be updated to 12 columns soon)
- Add a button to add a widget
- Add a button to remove a widget
- Add a button to duplicate a widget with the same content
- Create a simple and interactive widget with dynamic import (Module Federation)
- Create a complex widget that displays a service's performance graph with dynamic import (Module Federation)
- Handle responsiveness following screen size using breakpoints

How to test it?
- Go to `centreon/www/widgets/centreon-text-widget` then run `npm run build`
- Go to `centreon/www/widgets/centreon-performance-graph` then run `npm run build`
- In the dev ecosystem, check out to `custom-views-nightly`
- Start the dev ecosystem
- Go to "Home > Custom views (nightly)"
- Play with available widgets :)

About performances:

Here is a screenshot when dragging a widget without moving other widgets
![Screenshot 2022-10-19 at 11 58 59](https://user-images.githubusercontent.com/12515407/196661222-5aaf7e67-2d69-450d-9e5b-56e0210ae167.png)

Dragging a widget that moves widgets around
![Screenshot 2022-10-19 at 12 00 13](https://user-images.githubusercontent.com/12515407/196661390-65d9e77b-66f2-47bc-8090-f8929538f7ff.png)

Resizing the window to change the global layout
![Screenshot 2022-10-19 at 12 00 58](https://user-images.githubusercontent.com/12515407/196661711-44c8b1f1-e30e-41f1-8a09-7b0e51c92dd7.png)


## Type of change


## Target serie


<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

#### Community contributors & Centreon team
